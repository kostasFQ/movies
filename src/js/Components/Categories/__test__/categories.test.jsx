import React from 'react';
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { movie } from '../../../../../configs/jest/__mocks__/mockStore';
import ConnectedCategories from '../index';

const initialState = { movie };
const mStore = configureMockStore([thunk]);
const store = mStore(initialState);

const testRenderer = TestRenderer.create(
  <Provider store={store}><ConnectedCategories /></Provider>,
);
const testInstance = testRenderer.root;

test('1.test CONTENT component render', () => {
  const result = testRenderer.toJSON();
  expect(result.type).toBe('div');
  expect(result).toMatchSnapshot();
});

test('2.test CONTENT component render button click', () => {
 const e ={ target: { value: "popular" } }
  const but = testInstance.findByProps({value: "popular"})
  but.props.onClick(e);
});

test('3.test CONTENT component render select onChange', () => {
  const e ={ target: { value: "popular" } }
  const but = testInstance.findByType('select')
  but.props.onChange(e);
});


/*test('getCategoryMovie in CONTENT', async () => {
  await store.dispatch(getCategoryMovie('popular'));
  expect(store.getActions()[0].type).toBe('START_FETCH');
  expect(store.getActions()[1].type).toBe('SAVE_URL');
  expect(store.getActions()[2].type).toBe('GET_TRADING');
  expect(store.getActions()[3].type).toBe('FINISH_FETCH');
});

test('getDropdownMovie in CONTENT', async () => {
  await store.dispatch(getDropdownMovie(28));
  expect(store.getActions()[0].type).toBe('START_FETCH');
  expect(store.getActions()[1].type).toBe('SAVE_URL');
  expect(store.getActions()[2].type).toBe('GET_TRADING');
  expect(store.getActions()[3].type).toBe('FINISH_FETCH');
});*/
