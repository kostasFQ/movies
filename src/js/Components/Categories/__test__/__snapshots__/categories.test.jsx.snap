// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1.test CONTENT component render 1`] = `
<div
  className="container"
>
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="popular"
  >
    Trending
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="top_rated"
  >
    Top Rated
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="upcoming"
  >
    Coming Soon
  </button>
  <select
    className="item"
    onChange={[Function]}
  >
    <option
      value={28}
    >
      Action
    </option>
    <option
      value={12}
    >
      Adventure
    </option>
    <option
      value={16}
    >
      Animation
    </option>
    <option
      value={35}
    >
      Comedy
    </option>
    <option
      value={80}
    >
      Crime
    </option>
    <option
      value={99}
    >
      Documentary
    </option>
    <option
      value={18}
    >
      Drama
    </option>
    <option
      value={10751}
    >
      Family
    </option>
    <option
      value={14}
    >
      Fantasy
    </option>
    <option
      value={36}
    >
      History
    </option>
    <option
      value={27}
    >
      Horror
    </option>
    <option
      value={10402}
    >
      Music
    </option>
    <option
      value={9648}
    >
      Mystery
    </option>
    <option
      value={10749}
    >
      Romance
    </option>
    <option
      value={878}
    >
      Science Fiction
    </option>
    <option
      value={10770}
    >
      TV Movie
    </option>
    <option
      value={53}
    >
      Thriller
    </option>
    <option
      value={10752}
    >
      War
    </option>
    <option
      value={37}
    >
      Western
    </option>
  </select>
</div>
`;

exports[`CONTENT 1`] = `
<div
  className="container"
>
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="popular"
  >
    Trending
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="top_rated"
  >
    Top Rated
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="upcoming"
  >
    Coming Soon
  </button>
  <select
    className="item"
    onChange={[Function]}
  >
    <option
      value={28}
    >
      Action
    </option>
    <option
      value={12}
    >
      Adventure
    </option>
    <option
      value={16}
    >
      Animation
    </option>
    <option
      value={35}
    >
      Comedy
    </option>
    <option
      value={80}
    >
      Crime
    </option>
    <option
      value={99}
    >
      Documentary
    </option>
    <option
      value={18}
    >
      Drama
    </option>
    <option
      value={10751}
    >
      Family
    </option>
    <option
      value={14}
    >
      Fantasy
    </option>
    <option
      value={36}
    >
      History
    </option>
    <option
      value={27}
    >
      Horror
    </option>
    <option
      value={10402}
    >
      Music
    </option>
    <option
      value={9648}
    >
      Mystery
    </option>
    <option
      value={10749}
    >
      Romance
    </option>
    <option
      value={878}
    >
      Science Fiction
    </option>
    <option
      value={10770}
    >
      TV Movie
    </option>
    <option
      value={53}
    >
      Thriller
    </option>
    <option
      value={10752}
    >
      War
    </option>
    <option
      value={37}
    >
      Western
    </option>
  </select>
</div>
`;

exports[`test CONTENT component render 1`] = `
<div
  className="container"
>
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="popular"
  >
    Trending
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="top_rated"
  >
    Top Rated
  </button>
  <button
    className="item"
    onClick={[Function]}
    type="button"
    value="upcoming"
  >
    Coming Soon
  </button>
  <select
    className="item"
    onChange={[Function]}
  >
    <option
      value={28}
    >
      Action
    </option>
    <option
      value={12}
    >
      Adventure
    </option>
    <option
      value={16}
    >
      Animation
    </option>
    <option
      value={35}
    >
      Comedy
    </option>
    <option
      value={80}
    >
      Crime
    </option>
    <option
      value={99}
    >
      Documentary
    </option>
    <option
      value={18}
    >
      Drama
    </option>
    <option
      value={10751}
    >
      Family
    </option>
    <option
      value={14}
    >
      Fantasy
    </option>
    <option
      value={36}
    >
      History
    </option>
    <option
      value={27}
    >
      Horror
    </option>
    <option
      value={10402}
    >
      Music
    </option>
    <option
      value={9648}
    >
      Mystery
    </option>
    <option
      value={10749}
    >
      Romance
    </option>
    <option
      value={878}
    >
      Science Fiction
    </option>
    <option
      value={10770}
    >
      TV Movie
    </option>
    <option
      value={53}
    >
      Thriller
    </option>
    <option
      value={10752}
    >
      War
    </option>
    <option
      value={37}
    >
      Western
    </option>
  </select>
</div>
`;
